telmos_main.py

```diff
@@ -36,56 +24,253 @@ def telmos_main(delta_root, tmfs_root, tel_year, tel_id, tel_scenario,
     suffixes = range(3, 9)
     
     sr_array = []
-    all_names = []
     for suffix in suffixes:
         data = []
         names = []
         for period, purpose, mode in product(periods, purposes, modes):
             file_name = "%s_%s_%s_%s.txt" % (purpose, mode, period, suffix)
-            factor_file = os.path.join(factors_base, file_name)
-            try:
-                data.append(np.loadtxt(factor_file))
-                names.append(file_name)
-            except FileNotFoundError as f:
-                # If any file can not be found - abort
-                log_func("Could not find file %s\nAborting" % f)
-                return
+            factor_file = os.path.join(factors_dir, file_name)
+            data.append(np.loadtxt(factor_file))
+            names.append(file_name)
         sr_array.append(data)
-        all_names.append(names)
-    sr_array = np.asarray(sr_array)
+    return np.asarray(sr_array)
+
+def load_cte_tod_files(tod_files, cte_files, file_base):
+    tod_data = []
+    cte_data = []
+    for t_file, c_file in zip(tod_files, cte_files):
+        # TMfS14 had a long-distance module that required some CTE/TOD files to have
+        # _All appended to the end. This can now be removed if needed.
+        if (os.path.exists(os.path.join(file_base, t_file)) and
+               os.path.exists(os.path.join(file_base, c_file))):
+            t_file_name = t_file
+            c_file_name = c_file
+        else:
+            t_file_name = t_file.replace("_ALL","")
+            c_file_name = c_file.replace("_ALL","")
```

Comment:
Current set-up isn't flexible for if one file ends in `_ALL` and the other doesn't. How about:

```suggestion
        t_file_name = (
            t_file if os.path.isfile(os.path.join(file_base, t_file))
            else t_file.upper().replace("_ALL", "")
        )
        c_file_name = (
            c_file if os.path.isfile(os.path.join(file_base, c_file))
            else c_file.upper().replace("_ALL", "")
        )
```

This also:

- ensures our `_ALL` is properly replaced, by converting the filename to uppercase
- uses `os.path.isfile` rather than `os.path.exists`, so it can't be accidentally fooled by a folder with the name we're looking for.